src/core/src/op/search_sorted.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <openvino/op/search_sorted.hpp>
+
+#include "itt.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "search_sorted_shape_inference.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+
+SearchSorted::SearchSorted(const Output<Node>& sorted_sequence, const Output<Node>& values, bool right_mode)
+    : Op({sorted_sequence, values}),
+      m_right_mode(right_mode) {
+    constructor_validate_and_infer_types();
+}
+
+bool SearchSorted::validate() const {
+    NODE_VALIDATION_CHECK(this, get_input_size() == 2);
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0) == get_input_element_type(1),
+                          "Sorted sequence and values must have the same element type.");
+
+    const auto& sorted_shape = get_input_partial_shape(0);
+    const auto& values_shape = get_input_partial_shape(1);
+
+    if (sorted_shape.rank().is_static() && values_shape.rank().is_static() && sorted_shape.rank().get_length() > 1) {
```

Comment:
```suggestion
    if (sorted_shape.rank().is_static() && values_shape.rank().is_static() && sorted_shape.rank().get_length() > 0) {
```

Reply:
No, 1d case is a special case, that's why that check is added.